hsmd/hsmd.c

```diff
@@ -436,7 +437,7 @@ static struct io_plan *init_hsm(struct io_conn *conn,
 	struct secret *hsm_encryption_key;
 	struct bip32_key_version bip32_key_version;
 	u32 minversion, maxversion;
-	const u32 our_minversion = 2, our_maxversion = 3;
+	const u32 our_minversion = HSM_MIN_VERSION, our_maxversion = HSM_MAX_VERSION;
```

Comment:
I don't think using the `HSM_{MIN,MAX}_VERSION` constants is correct here.  The versioned protocol is between CLN daemons (`lightningd`, `channeld`, `gossipd`, ...) and the HSMD represented here.

The constants `HSM_MIN_VERISON` and `HSM_MAX_VERSION` are for the daemons **other** than the `hsmd`.  We need to increase the `HSM_MIN_VERSION` for `gossipd` to 5 because it must use the `hsmd_sign_any_cannouncement_req` message which is only available in version 5 and later.

But the native `hsmd` (represented by this code) is capable of fulfilling previous versions of the protocol.  I think the correct minimum version is 4 (the first version that supports `hsmd_init_reply_v4` with `hsm_capabilities`).

Reply:
I agree, thanks for this good description